### PR TITLE
fix tomcat-10x-plugin and add test case to support tomcat7.x-8.x-9.x

### PR DESCRIPTION
**Pull request by xzyJavaX**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by xzyJavaX** [comment]
https://github.com/apache/skywalking-java/pull/286/files#r1119753555

**Comment by wu-sheng** [comment]
Somehow Spring 6 case fails? Could you recheck?

**Comment by lujiajing1126** [comment]
> Somehow Spring 6 case fails? Could you recheck?

The only issue I could find is that we expected `POST:/create` call returns status code 200, but in the actual data is 201.

But I don't think this PR is relevant since for `EntrySpan`, the tag is set when `stackDepth == currentMaxDepth`. This means only the Spring plugin can set tags.

@kezhenxu94 @wu-sheng any idea?

**Comment by kezhenxu94** [comment]
> > Somehow Spring 6 case fails? Could you recheck?
> 
> The only issue I could find is that we expected `POST:/create` call returns status code 200, but in the actual data is 201.
> 
> But I don't think this PR is relevant. Since for `EntrySpan`, the tag is set when `stackDepth == currentMaxDepth`. This means only the Spring plugin can set tags.


This is somewhat relevant, I tested Spring 6 and expected the status code to be 201 but it was always 200, I was sure it was not an issue of Spring plugin but didn't dig deeper to the cause, so this PR accidentally fixes the bug too, so, please update the expected data for Spring 6, change status code from 200 to 201.

File: apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java [hunk 1]
```diff
@@ -82,9 +75,9 @@ public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allA
         HttpServletResponse response = (HttpServletResponse) allArguments[1];
 
         AbstractSpan span = ContextManager.activeSpan();
-        if (IS_SERVLET_GET_STATUS_METHOD_EXIST && response.getStatus() >= 400) {
+        Tags.HTTP_RESPONSE_STATUS_CODE.set(span, response.getStatus());
+        if (response.getStatus() >= 400) {
```
Comment (kezhenxu94) [inline apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java:79]: I think this fixes the issue I mentioned

Comment (lujiajing1126) [reply apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java:79]: > I think this fixes the issue I mentioned

But in the `EntrySpan`,

```java
    @Override
    public EntrySpan tag(String key, String value) {
        if (stackDepth == currentMaxDepth || isInAsyncMode) {
            super.tag(key, value);
        }
        return this;
    }
```

I suppose in the Tomcat plugin, the condition cannot be true. Do I miss something?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java:79]: It is better to debug this logic.

**Comment by wu-sheng** [comment]
What do these 201 and 204 mean actually?

**Review by wu-sheng — APPROVED** [review]
I am fine with Tomcat 10 part. Wait for @kezhenxu94

**Review by kezhenxu94 — APPROVED** [review]
lgtm

File: apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java (hunk 1 above)
Comment (lujiajing1126) [reply apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java:79]: image

After debugging, it seems that the above method is not called. Instead, `public AbstractTracingSpan tag(AbstractTag<?> tag, String value)` is called while `EntrySpan` does not override this method (But `ExitSpan` does!). Then the `http.status_code` tag is set.

I am not sure if it is a feature or a bug? @wu-sheng @kezhenxu94

Comment (lujiajing1126) [reply apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java:79]: Another question is why Tomcat and Spring set different status code...

**Pull request merged**